forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-gateway/forgerock-openbanking-uk-aspsp-rs-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/rs/api/payment/verifier/DetachedJwsVerifier.java

```diff
@@ -104,22 +112,26 @@ private boolean isB64ClaimHeaderSetToFalse(String jwsDetachedSignature, String p
         String jwsSerialized = rebuildJWS(jwsDetachedSignature, payload);
         log.debug("The reconstructed JWS from the detached signature: {}", jwsSerialized);
         SignedJWT jws = (SignedJWT) JWTParser.parse(jwsSerialized);
-        if (jws.getHeader().getCustomParam("b64") != null) {
-            log.debug("B64 claim header in the detached signature: {}", jws.getHeader().getCustomParam("b64"));
-            Object b64Param = jws.getHeader().getCustomParam("b64");
-            if (!(b64Param instanceof Boolean)) {
-                return false;
-            }
-            return ((Boolean)b64Param) == false;
-        }
-        return false;
+        boolean isSetToFalse = jws.getHeader().isBase64URLEncodePayload();
+        return !isSetToFalse;
     }
 
     private boolean isB64ClaimHeaderPresent(String jwsDetachedSignature, String payload) throws OBErrorException, ParseException {
         String jwsSerialized = rebuildJWS(jwsDetachedSignature, payload);
         log.debug("The reconstructed JWS from the detached signature: {}", jwsSerialized);
         SignedJWT jws = (SignedJWT) JWTParser.parse(jwsSerialized);
-        return jws.getHeader().getCustomParam("b64") != null;
+        Set<String> params = jws.getHeader().getIncludedParams();
+        return params.contains("b64");
+    }
+
+
+    private boolean newIsB64ClaimHeaderPresent(String jwsDetachedSignature, String payload) throws OBErrorException {
+        String jwsSerialised = rebuildJWS(jwsDetachedSignature, payload);
+        String[] splitJws = jwsSerialised.split("\\.");
+        Base64.Decoder b64Decoder = Base64.getDecoder();
+        byte[] jsonBytes = b64Decoder.decode(splitJws[0]);
+        String jsonString = new String(jsonBytes, StandardCharset.UTF_8);
+        return jsonString.contains("\"b64\":");
```

Comment:
`headerBytes` and `headerJson` might aid readability?

Or personally, I would do away with `headerBytes` and just construct `headerJson` on one line :-)